pytorch_lightning/lite/wrappers.py

```diff
@@ -102,12 +108,21 @@ def _convert_float_tensor(t: Tensor) -> Tensor:
         args, kwargs = apply_to_collection([args, kwargs], function=_convert_float_tensor, dtype=Tensor)
 
         with self._precision_plugin.forward_context():
-            output = self.module(*args, **kwargs)
+            output = self._forward_module(*args, **kwargs)
 
         to_type = torch.get_default_dtype()
         output = apply_to_collection(output, function=_convert_float_tensor, dtype=Tensor)
         return output
 
+    def __getattr__(self, item: Any) -> Any:
+        try:
+            # __getattr__ gets called as a last resort if the attribute does not exist
+            # call nn.Module's implementation first
+            return super().__getattr__(item)
+        except AttributeError:
```

Comment:
Why the try-except instead of `hasattr`?

Reply:
Afaik `hasattr` is implemented by calling `getattr` and catching `AttributeError`. So if you would call `hasattr` here, `getattr` would call this very method again and you'd end up with recursion errors since you can only call this on an instance (`self`) and not the superclass

Reply:
Yes exactly, that is the reason. 
However, there is still something weird going on with one of the failing deepspeed test and haven't figured it out yet. 

```
tests/lite/test_lite.py:459: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytorch_lightning/lite/lite.py:405: in _run_impl
    return self._strategy.launcher.launch(run_method, *args, **kwargs)
pytorch_lightning/strategies/launchers/subprocess_script.py:93: in launch
    return function(*args, **kwargs)
pytorch_lightning/lite/lite.py:412: in _run_with_strategy_setup
    return run_method(*args, **kwargs)
tests/lite/test_lite.py:402: in run
    model, optimizer = self.setup(model, optimizer)
pytorch_lightning/lite/lite.py:174: in setup
    model = _LiteModule(model, self._precision_plugin, original_module=original_model)
/usr/local/lib/python3.7/dist-packages/deepspeed/runtime/zero/partition_parameters.py:249: in wrapper
    if not hasattr(module, "_ds_child_entered"):
pytorch_lightning/lite/wrappers.py:124: in __getattr__
    return getattr(self._original_module, item)
pytorch_lightning/lite/wrappers.py:124: in __getattr__
    return getattr(self._original_module, item)
pytorch_lightning/lite/wrappers.py:124: in __getattr__
    return getattr(self._original_module, item)
E   RecursionError: maximum recursion depth exceeded while calling a Python object
!!! Recursion detected (same locals & position)
```

My suspicion is that deepspeed does some patching which I haven't found yet.

Reply:
Ok, finally figured this out. Deepspeed patches the init globally for all nn Modules. That wrapper called hasattr which then called our getattr and at this point `_original_module ` wasn't defined yet so it would recurse infinitely.